src/rocdecode/roc_decoder_caps.h

```diff
@@ -29,80 +29,214 @@ THE SOFTWARE.
+                unsigned int attr_count;
+                std::vector<VASurfaceAttrib> attr_list;
+                va_config_attrib.type = VAConfigAttribRTFormat;
+                CHECK_VAAPI(vaGetConfigAttributes(va_display, profile_list[i], VAEntrypointVLD, &va_config_attrib, 1));
+                CHECK_VAAPI(vaCreateConfig(va_display, profile_list[i], VAEntrypointVLD, &va_config_attrib, 1, &va_config_id));
+                CHECK_VAAPI(vaQuerySurfaceAttributes(va_display, va_config_id, 0, &attr_count));
+                attr_list.resize(attr_count);
+                CHECK_VAAPI(vaQuerySurfaceAttributes(va_display, va_config_id, attr_list.data(), &attr_count));
+                for (int k = 0; k < attr_count; k++) {
+                    switch (attr_list[k].type) {
+                    case VASurfaceAttribPixelFormat:
+                    {
+                        switch (attr_list[k].value.value.i) {
+                            case VA_FOURCC_NV12:
+                                decode_cap_list_[j].output_format_mask |= 1 << rocDecVideoSurfaceFormat_NV12;
+                                break;
+                            case VA_FOURCC_P016:
+                                decode_cap_list_[j].output_format_mask |= 1 << rocDecVideoSurfaceFormat_P016;
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                        break;
+                    case VASurfaceAttribMinWidth:
+                        if (decode_cap_list_[j].min_width == 0 || (decode_cap_list_[j].min_width > 0 && decode_cap_list_[j].min_width > attr_list[k].value.value.i)) {
+                            decode_cap_list_[j].min_width = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMinHeight:
+                        if (decode_cap_list_[j].min_height == 0 || (decode_cap_list_[j].min_height > 0 && decode_cap_list_[j].min_height > attr_list[k].value.value.i)) {
+                            decode_cap_list_[j].min_height = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMaxWidth:
+                        if (decode_cap_list_[j].max_width < attr_list[k].value.value.i) {
+                            decode_cap_list_[j].max_width = attr_list[k].value.value.i;
+                        }
+                        break;
+                    case VASurfaceAttribMaxHeight:
+                        if (decode_cap_list_[j].max_height < attr_list[k].value.value.i) {
+                            decode_cap_list_[j].max_height = attr_list[k].value.value.i;
+                        }
+                        break;
+                    default:
+                        break;
+                    }
+                }
+            }
+        }
+
+        initialized_ = true;
```

Comment:
call `vaDestroyConfig(va_display, va_config_id)`; and  `vaTerminate(va_display);` at the end of this function.

Reply:
Yep thanks for catching this. Was going to add but somehow forgot.